modules/oe_editorial_unpublish/tests/Functional/CorporateWorkflowUnpublishTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_editorial_unpublish\Functional;
+
+use Drupal\Core\Url;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the unpublishing form for nodes.
+ */
+class EditorialUnpublishTest extends BrowserTestBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\node\NodeStorageInterface
+   */
+  protected $nodeStorage;
+
+  /**
+   * The test node.
+   *
+   * @var \Drupal\node\NodeInterface
+   */
+  protected $node;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $user;
+
+  protected static $modules = [
+    'user',
+    'node',
+    'system',
+    'oe_editorial',
+    'oe_editorial_corporate_workflow',
+    'oe_editorial_workflow_demo',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $this->nodeStorage = $entity_type_manager->getStorage('node');
+    $this->node = $this->nodeStorage->create(
+      [
+        'type' => 'oe_workflow_demo',
+        'title' => 'My node',
+        'moderation_state' => 'draft',
+      ]
+    );
+    $this->node->save();
+
+    /** @var \Drupal\user\RoleInterface $role */
+    $role = $entity_type_manager->getStorage('user_role')->load('oe_validator');
+    $this->user = $this->drupalCreateUser($role->getPermissions());
+    $this->drupalLogin($this->user);
+  }
+
+  /**
+   * Tests access to the unpublishing form.
+   */
+  public function testUnpublishAccess(): void {
+    $unpublish_url = Url::fromRoute('entity.node.unpublish', [
+      'node' => $this->node->id(),
+    ]);
```

Comment:
Add here a node that's bundle is not using content moderation and assert that you can't access the unpublish page.

Reply:
Updated tests